### PR TITLE
eliminate the difference between when having no RTT sample and having one

### LINKED ISSUE #3524: Inconsistency between early timeout calculations
For each connection, the loss detection timeout is defined as 2 * InitialRTT.

I wonder if this should be 3 * InitialRTT instead, because, once you have one RTT sample, PTO is calculated as 3 * InitialRTT.

I do not think we have a reason to believe that the variance of RTT (which applies as a multiplier here) increases when we obtain the first sample?

PS. The other benefit of making these two consistent is that we can get rid of `if (smoothed_rtt == 0)` in SetLossDetectionTimer. It would be possible to always rely on PTO calculation.

### PR DESCRIPTION
**Pull request by kazuho**
As pointed out in #3524, I think that we should assume same amount of RTT variance when there is no RTT sample and when there is one sample, because variance is unknown in both cases.

This PR makes that change while keeping the initial PTO being the same (i.e. 1 second), and slightly simplifies the timeout logic.

Fixes #3524.

### CODE REVIEW

File: draft-ietf-quic-recovery.md [hunk 1]
```diff
@@ -353,14 +353,17 @@ endpoint:
...
-smoothed_rtt = latest_rtt
-rttvar = latest_rtt / 2
+smoothed_rtt = rtt_sample
+rttvar = rtt_sample / 2
 ~~~
 
+where rtt_sample is either the initial RTT when there is no sample, or the
+latest RTT when there is one.
+
 On subsequent RTT samples, smoothed_rtt and rttvar evolve as follows:
```
Comment (ianswett) [inline draft-ietf-quic-recovery.md]: I'm concerned this section is now potentially ambiguous.  I believe we want to maintain the property that the first actual RTT measurement replaces the initialized values, and I don't think that's clear enough here.

File: draft-ietf-quic-recovery.md [hunk 2]
```diff
@@ -353,14 +353,17 @@ endpoint:
 On the first RTT sample for a network path, the smoothed_rtt is set to the
 latest_rtt.
```
Comment (ianswett) [inline draft-ietf-quic-recovery.md]: Moving this down above "On subsequent RTT samples," and adding that rttvar is also set on the first RTT sample  would help clarify this.

**Comment by kazuho** [comment]
@ianswett Thank you for the review. Would you ming checking if the updated text has what you have suggested?

File: draft-ietf-quic-recovery.md [hunk 3]
```diff
@@ -350,17 +350,19 @@ endpoint:
...
-smoothed_rtt = latest_rtt
-rttvar = latest_rtt / 2
+smoothed_rtt = rtt_sample
+rttvar = rtt_sample / 2
 ~~~
 
+where rtt_sample is the initial RTT.
+
+On the first RTT sample for the network path, smoothed_rtt and rttvar are set to
+the values using the formula above, using that first RTT sample as rtt_sample.
```
Comment (ianswett) [inline draft-ietf-quic-recovery.md]:
```suggestion
the values using the formula above, using that first RTT sample as rtt_sample.
This ensures a measurement erases the history of any persisted or default value.
```

Comment (kazuho) [reply draft-ietf-quic-recovery.md]: Thank you for the suggestion. Applied.

File: draft-ietf-quic-recovery.md [hunk 4]
```diff
@@ -350,17 +350,20 @@ endpoint:
...
+This ensures a measurement erases the history of any persisted or default value.
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]:
```suggestion
This ensures that the first measurement erases the history of any persisted or default value.
```

Comment (kazuho) [reply draft-ietf-quic-recovery.md]: Thank you for the suggestion. Applied in c4ed9aa.

File: draft-ietf-quic-recovery.md [hunk 5]
```diff
@@ -350,17 +350,21 @@ endpoint:
   min_rtt.  This limits the underestimation that a misreporting peer can cause
   to the smoothed_rtt.
 
-On the first RTT sample for a network path, the smoothed_rtt is set to the
-latest_rtt.
-
-smoothed_rtt and rttvar are computed as follows, similar to {{?RFC6298}}.  On
-the first RTT sample for a network path:
+smoothed_rtt and rttvar are computed as follows, similar to {{?RFC6298}}.  When
+there are no samples for a network path:
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]:
```suggestion
smoothed_rtt and rttvar are computed as follows, similar to {{?RFC6298}}.

When there are no samples for a network path, and on the first RTT sample for
the network path:
```

File: draft-ietf-quic-recovery.md [hunk 6]
```diff
@@ -350,17 +350,21 @@ endpoint:
...
+smoothed_rtt = rtt_sample
+rttvar = rtt_sample / 2
 ~~~
 
+where rtt_sample is the initial RTT.
+
+On the first RTT sample for the network path, smoothed_rtt and rttvar are set to
+the values using the formula above, using that first RTT sample as rtt_sample.
+This ensures that the first measurement erases the history of any persisted or
+default value.
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]:
```suggestion
Before any RTT samples are available, the initial RTT is used as rtt_sample. 
On the first RTT sample for the network path, that sample is used as rtt_sample.
This ensures that the first measurement erases the history of any persisted or
default values.
```

File: draft-ietf-quic-recovery.md [hunk 7]
```diff
@@ -517,12 +521,10 @@ data.
...
-The initial probe timeout for a new connection or new path SHOULD be
-set to twice the initial RTT.  Resumed connections over the same network
-SHOULD use the previous connection's final smoothed RTT value as the resumed
-connection's initial RTT.  If no previous RTT is available, the initial RTT
-SHOULD be set to 500ms, resulting in a 1 second initial timeout as recommended
-in {{?RFC6298}}.
+When no previous RTT is available, the initial RTT SHOULD be set to 333ms,
+resulting in a 1 second initial timeout as recommended in {{?RFC6298}}. Resumed
+connections over the same network SHOULD use the previous connection's final
+smoothed RTT value as the resumed connection's initial RTT.
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]:
```suggestion
Resumed
connections over the same network SHOULD use the previous connection's final
smoothed RTT value as the resumed connection's initial RTT. When no previous
RTT is available, the initial RTT SHOULD be set to 333ms, resulting in a 1 second
initial timeout, as recommended in {{?RFC6298}}.
```

File: draft-ietf-quic-recovery.md (hunk 5 above)
Comment (kazuho) [reply draft-ietf-quic-recovery.md]: Splitting the text when having no samples and having one is intentional, see the changes suggested in https://github.com/quicwg/base-drafts/pull/3525#pullrequestreview-374825664.

I'm personally fine with the changes proposed here, but I would like to see what @ianswett thinks. Maybe it's OK, as the we have two paragraphs right after, discussing about each of the two conditions.

Comment (ianswett) [reply draft-ietf-quic-recovery.md]: I think @janaiyengar suggestion seems good and slightly clearer than before, though maybe line 361("where rtt_sample is the initial RTT." needs to be updated too?

File: draft-ietf-quic-recovery.md [hunk 8]
```diff
@@ -517,12 +521,11 @@ data.
 
 ### Handshakes and New Paths
 
-The initial probe timeout for a new connection or new path SHOULD be
-set to twice the initial RTT.  Resumed connections over the same network
-SHOULD use the previous connection's final smoothed RTT value as the resumed
-connection's initial RTT.  If no previous RTT is available, the initial RTT
-SHOULD be set to 500ms, resulting in a 1 second initial timeout as recommended
-in {{?RFC6298}}.
+Resumed
```
Comment (ianswett) [inline draft-ietf-quic-recovery.md]: nit: reflow the text

**Comment by kazuho** [comment]
@janaiyengar @ianswett Thank you for the suggestions. I've applied them.

**Pull request merged**